rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -436,15 +447,22 @@ void debugger_frame::OnSelectUnit()
 	if (m_choice_units->count() < 1) return;
 
 	const auto weak = m_choice_units->currentData().value<std::weak_ptr<cpu_thread>>();
+	const auto render = m_choice_units->currentData().value<rsx::thread*>();
 
-	if (!weak.owner_before(cpu) && !cpu.owner_before(weak))
+	if (!render && !weak.owner_before(cpu) && !cpu.owner_before(weak))
 	{
 		// They match, nothing to do.
 		return;
 	}
 
+	if (render && render == (!Emu.IsStopped() ? m_rsx : nullptr))
```

Comment:
Again with the unreadable code...
```suggestion
	if (render && !Emu.IsStopped() && render == m_rsx)
```

Reply:
Fixed.